### DISCUSSION TITLE
Save the full page with the model when downloaded, so it can be loaded offline.

### DISCUSSION THREAD

**Discussion by mx** [BlafKing/sd-civitai-browser-plus#171] [Ideas]
This is a fantastically useful extension, and I often end up checking the model's page for information on how to use the model. Way more than I actually search civitai. So it would be really helpful if checking on the information for models you already have didn't depend on Civitai or your own connection being up. It would also ensure that the Civitai page won't change out from under you.

**Comment by mx** [comment]
Looking at it again, it does save the full page with the model when you request to save the model info, so that's great! It really just needs to (possibly optionally?) load the saved HTML file when you click the Civitai icon on the model card, rather than re-pulling it from Civitai itself.

**Comment by BlafKing** [comment]
Thanks for the request!
This is now possible in the latest version!

A small change is that `Save images` button no longer saves the .html and API info now, the `Save model info` button now does this instead.
Another small change is that the .html file now automatically gets saved after a download if the `Save info after download` option is on in the filter dropdown.

There is now a settings option in SD-WebUI settings tab under the CivitAI tab which can be used to use any local .HTML files when pressing the model card CivitAI button.
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/9644716/ec98f2ee-8337-47ac-b239-f5796485f364)

This will then attempt to find a local .html file, and will only send an API request for the info if the file has not been found! :)
